### ISSUE TITLE
Sort order on bed file

### ISSUE THREAD

**Issue by skitchen19** [bcgsc/LongStitch#43]
I am running into an endless runtime on the "Finding Breakpoints" step of tigmint-cut. Going on 6 hrs for a 300M genome. Based on the previous resolved issues regarding this problem, it would appear that sort order may be the issue. All packages and dependencies are otherwise up to date.

My scaffold names are alphanumeric, i.e. scaffold1, scaffold2, etc. The current default sort order puts scaffold1, scaffold10, and then scaffold2. It that how it should be? If not, would this cause a slow down at this step? Also, does the bed file order need to match the genome index (.fai) order?

Thanks!

**Comment by lcoombe** [comment]
Hi @skitchen19,

Yes, you're right that that runtime for that step of Tigmint is much longer than expected - I would expect that step to run quite quickly for a genome of that size. Could you report the version of `sort` that you are using on your machine?
The naming of the scaffolds shouldn't matter, nor should the relation to the `.fai` file. I seem to recall previously seeing some funny issues with older versions of `sort`, so I'd be interested to see that version.

Also, have you successfully run the assembly demo (in `test` directory)? If not, you could try running that and comparing the outputs to the files in `expected_outputs` to ensure that the steps were working as expected, and the installation is working properly.

Thank you for your interest in LongStitch!
Lauren

**Comment by skitchen19** [comment]
Thank for getting back to me so quickly.

The version of `sort` is 8.22. 

I was able to get through the test data without encountering this problem.

**Comment by lcoombe** [comment]
Hi @skitchen19,
Can you confirm that the first test in the demo (using `test_scaffolds1.fa`) did in fact make a cut in the contigs? Here's the expected contents of the file `test_scaffolds1.test_reads1.cut250.molecule.size2000.distauto.trim0.window1000.spanauto.breaktigs.fa.bed`:
```
103642	0	1129	103642
185898	0	6886	185898
190623	0	62735	190623
180793	0	8001	180793-1
180793	8001	8024	180793-2
180793	8024	12931	180793-3
187066	0	3972	187066
181193	0	23261	181193
187190	0	22186	187190
182485	0	7490	182485
187264	0	3700	187264
182499	0	14036	182499
187638	0	1217	187638
183815	0	22391	183815
189924	0	39009	189924
185554	0	3448	185554
189963	0	118605	189963
185693	0	82911	185693
190275	0	56919	190275
```

Could you try updating the `sort` version that you're using? You could do that by installing `coreutils` with conda, for example.

**Comment by skitchen19** [comment]
Hi @lcoombe,

Sorry it has taken me a while to return to this analysis. I updated my version of `sort` to 8.32 and re-ran the test script. The name of my output file was slightly different than yours, `test_scaffolds1.test_reads1.cut250.molecule.size2000.trim0.window1000.spanauto.breaktigs.fa.bed`, but matched your file 
 output above. 
```
103642  0       1129    103642
185898  0       6886    185898
190623  0       62735   190623
180793  0       8001    180793-1
180793  8001    8024    180793-2
180793  8024    12931   180793-3
187066  0       3972    187066
181193  0       23261   181193
187190  0       22186   187190
182485  0       7490    182485
187264  0       3700    187264
182499  0       14036   182499
187638  0       1217    187638
183815  0       22391   183815
189924  0       39009   189924
185554  0       3448    185554
189963  0       118605  189963
185693  0       82911   185693
190275  0       56919   190275
```
I did not detect any errors during the test run but when I compare the final scaffolds with `abyss-fac` they are not identical. For example my output:
```
abyss-fac test_scaffolds1.k24.w100.tigmint-ntLink.longstitch-scaffolds.fa
n       n:500   L50     min     N75     N50     N25     E-size  max     sum     name
3       2       1       4907    477742  477742  477742  472934  477742  482649  test_scaffolds1.k24.w100.tigmint-ntLink.longstitch-scaffolds.fa
```
compared to the expected output:
```
abyss-fac ./expected_outputs/test_scaffolds1.k24.w100.tigmint-ntLink.longstitch-scaffolds.fa
n       n:500   L50     min     N75     N50     N25     E-size  max     sum     name
3       2       1       4907    462255  462255  462255  457451  462255  467162  ./expected_outputs/test_scaffolds1.k24.w100.tigmint-ntLink.longstitch-scaffolds.fa
```
Is that expected?

**Comment by lcoombe** [comment]
Hi @skitchen19,

Thanks for pointing out the file naming difference - I updated the expected output files with the newest version of each of the tools.

Are you using the most up-to-date versions of Tigmint and ntLink? I was able to reproduce your stats, but using an older version of ntLink (v1.0.1). The expected outputs are for the more recent releases of each of the tools (and I'll endeavour to keep that updated as we do new releases going forward)
